qt4 build fails during installPhase [Darwin]

## Issue description
qt4 fails during the install phase
[[hydra](https://hydra.nixos.org/build/58837766)]
```
make[1]: Entering directory '/private/tmp/nix-build-qt-4.8.7.drv-0/qt-everywhere-opensource-src-4.8.7/src/corelib'
make -f Makefile.Release install
make[2]: Entering directory '/private/tmp/nix-build-qt-4.8.7.drv-0/qt-everywhere-opensource-src-4.8.7/src/corelib'
"/nix/store/yc00923s1wacd726fwf5yn9nkxbkfkxy-qt-4.8.7/lib/QtCore.framework/Contents/MacOS/QtCore"
/nix/store/p436632iqmvvzn2m0rg5wj270xr9niap-bash-4.4-p12/bin/bash: /nix/store/yc00923s1wacd726fwf5yn9nkxbkfkxy-qt-4.8.7/lib/QtCore.framework/Contents/MacOS/QtCore: No such file or directory
make[2]: *** [Makefile.Release:29195: install_target] Error 127
make[2]: Leaving directory '/private/tmp/nix-build-qt-4.8.7.drv-0/qt-everywhere-opensource-src-4.8.7/src/corelib'
make[1]: *** [Makefile:46: release-install] Error 2
make[1]: Leaving directory '/private/tmp/nix-build-qt-4.8.7.drv-0/qt-everywhere-opensource-src-4.8.7/src/corelib'
make: *** [Makefile:226: sub-corelib-install_subtargets-ordered] Error 2
note: keeping build directory ‘/private/tmp/nix-build-qt-4.8.7.drv-0’
builder for ‘/nix/store/kb7j85g6h8458a93rp0bvhfgaczphdc7-qt-4.8.7.drv’ failed with exit code 2
error: build of ‘/nix/store/kb7j85g6h8458a93rp0bvhfgaczphdc7-qt-4.8.7.drv’ failed
```
The file in question is built but the install phase seems to be depending on the nix-store location before it is copied there. 

```
ls /private/tmp/nix-build-qt-4.8.7.drv-0/qt-everywhere-opensource-src-4.8.7/lib
...
QtCore.framework
...
```

There also seem to be a few possibly erroneous file deletions by the build system:
```
rm -f QtCore QtCore.framework/QtCore ../../lib/QtCore.framework/QtCore
...
mv -f QtCore ../../lib/QtCore.framework/Versions/4/QtCore
ln -s Versions/4/QtCore ../../lib/QtCore.framework/QtCore
rm -f ../../lib/QtCore.framework/Versions/Current
ln -s 4 ../../lib/QtCore.framework/Versions/Current
...
```

### Steps to reproduce
`nix-build '<nixpkgs>' -A qt4 -K`


## Technical details

* System: (MacOS 10.11)
* Nix version: (`1.11.13`)
* Nixpkgs version: (`17.09pre-git`)
* Sandboxing enabled: (`relaxed`)

@vcunat

--- a/pkgs/development/libraries/qt-4.x/4.8/default.nix
+++ b/pkgs/development/libraries/qt-4.x/4.8/default.nix
@@ -183,10 +183,38 @@ stdenv.mkDerivation rec {
     sed -i 's/^\(LIBS[[:space:]]*=.*$\)/\1 -lobjc/' ./src/corelib/Makefile.Release
   '';
 
-  postInstall =
-    ''
+  installPhase = optionalString stdenv.isDarwin ''
+    runHook preInstall
+    cp -r lib $out
+
+    mkdir -p $out/Applications
+    mv bin/*.app $out/Applications
+    rm -rf bin/*.app
+
+    cp -r bin $out
+
+    mkdir -p $out/share/doc/${name}
+    mkdir -p $out/lib
+    mkdir -p $out/lib/qt4/plugins
+    mkdir -p $out/lib/qt4/imports
+    mkdir -p $out/bin
+    mkdir -p $out/include
+    mkdir -p $out/share/${name}
+
+    cp -r mkspecs $out/share/${name}
+    cp -r translations $out/share/${name}
+    cp -r tools/linguist/phrasebooks $out/share/${name}
+    cp tools/porting/src/q3porting.xml $out/share/${name}
+
+    cp -r plugins $out/lib/qt4
+    cp -r imports $out/lib/qt4
+    cp -r doc/* $out/share/doc/${name}
+    runHook postInstall
+  '';
+
+  postInstall = optionalString (!stdenv.isDarwin) ''
       rm -rf $out/tests
-    '';
+  '';
 
   crossAttrs = {
     # I've not tried any case other than i686-pc-mingw32.

